### ISSUE TITLE
Dimension mismatch in `powf` operation for tensors

### ISSUE THREAD

**Issue by L-M-Sherlock** [tracel-ai/burn#2387]
**Describe the bug**

When attempting to perform a `powf` operation between two tensors of different shapes, the operation fails with a dimension mismatch error instead of broadcasting the smaller tensor.

**To Reproduce**

1. Create a tensor `a` with shape [3]
2. Create a tensor `b` with shape [1]
3. Attempt to perform `a.powf(b)`

Code Example:

```rust
    #[test]
    fn pow() {
        let a = Tensor::from_floats([2.0, 3.0, 4.0], &NdArrayDevice::Cpu);
        let b = Tensor::from_floats([1.0], &NdArrayDevice::Cpu);
        let c = a.powf(b);
        dbg!(&c);
    }
```

You will get:

```
Zip: Producer dimension mismatch, expected: IxDynImpl(Inline(1, [3, 0, 0, 0])), got: IxDynImpl(Inline(1, [1, 0, 0, 0]))
```

**Expected behavior**

The operation should broadcast the smaller tensor `b` to match the shape of `a`, resulting in a successful `powf` operation.

**Environment**

burn 0.13.1

**Additional context**

You may ask why not use `powf_scalar`. Here is a case to explain why I cannot use it:

```rust
    #[test]
    fn pow() {
        let a = Tensor::from_floats([2.0, 3.0, 4.0], &NdArrayDevice::Cpu);
        let b = Param::from_tensor(Tensor::from_floats([1.0], &NdArrayDevice::Cpu));
        let c = (a.log() * b.get(0)).exp();
        let d = Tensor::from_floats([2.0, 3.0, 4.0], &NdArrayDevice::Cpu);
        let loss = c - d;
        let gradients = loss.backward();
        let grad = b.grad(&gradients);
        dbg!(&grad);
    }
```

The result is:

```
&grad = Some(
    Tensor {
        primitive: NdArrayTensor {
            array: [10.227308], shape=[1], strides=[1], layout=CFcf (0xf), dynamic ndim=1,
        },
    },
)
```

But if I use `powf_scalar`:

```rust
    #[test]
    fn pow() {
        let a = Tensor::from_floats([2.0, 3.0, 4.0], &NdArrayDevice::Cpu);
        let b = Param::from_tensor(Tensor::from_floats([1.0], &NdArrayDevice::Cpu));
        let c = a.powf_scalar(b.get(0).into_scalar());
        let d = Tensor::from_floats([2.0, 3.0, 4.0], &NdArrayDevice::Cpu);
        let loss = c - d;
        let gradients = loss.backward();
        let grad = b.grad(&gradients);
        dbg!(&grad);
    }
```

The result is:

```
&grad = None
```

So the gradient is lost in `powf_scalar`.

**Related issue and PR**
- #590
- #748
- https://github.com/open-spaced-repetition/fsrs-rs/pull/240

**Comment by laggui** [comment]
Thanks for flagging this! I just opened a PR that should resolve the problem.

**Comment by L-M-Sherlock** [comment]
Thanks for fixing it!
